data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusSinkService.java

```diff
@@ -4,49 +4,415 @@
+            labelBuilder.setName(key).setValue(value.toString());
+            Types.Label label = labelBuilder.build();
+            arrayList.add(label);
+        }
+    }
+
+    private static String sanitizeName(String name) {
+        return BODY_PATTERN
+                .matcher(PREFIX_PATTERN.matcher(name).replaceFirst("_"))
+                .replaceAll("_");
+    }
+
+    private static long getTimeStampVal(String time) {
+        LocalDateTime localDateTimeParse = LocalDateTime.parse(time,
+                DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS'Z'"));
+        LocalDateTime localDateTime = LocalDateTime.parse(localDateTimeParse.toString());
+        ZonedDateTime zdt = ZonedDateTime.of(localDateTime, ZoneId.systemDefault());
+        return zdt.toInstant().toEpochMilli();
+    }
+
+    /**
+     * * This method logs Failed Data to DLQ and Webhook
+     *  @param endPointResponses HttpEndPointResponses.
+     */
+    private void logFailedData(final HttpEndPointResponse endPointResponses) {
+        FailedDlqData failedDlqData =
+                FailedDlqData.builder()
+                        .withUrl(endPointResponses.getUrl())
+                        .withMessage(endPointResponses.getErrorMessage())
+                        .withStatus(endPointResponses.getStatusCode()).build();
+        LOG.info("Failed to push the data. Failed DLQ Data: {}",failedDlqData);
+
+        logFailureForDlqObjects(failedDlqData);
+    }
+
+    private void releaseEventHandles(final boolean result) {
+        for (EventHandle eventHandle : bufferedEventHandles) {
+            eventHandle.release(result);
+        }
+        bufferedEventHandles.clear();
+    }
+
+    /**
+     * * This method pushes bufferData to configured HttpEndPoints
+     *  @param data byte[] data.
+     */
+    private HttpEndPointResponse pushToEndPoint(final byte[] data) throws IOException {
+        HttpEndPointResponse httpEndPointResponses = null;
+        final ClassicRequestBuilder classicHttpRequestBuilder =
+                httpAuthOptions.get(prometheusSinkConfiguration.getUrl()).getClassicHttpRequestBuilder();
+
+        final byte[] compressedBufferData = Snappy.compress(data);
+        HttpEntity entity = new ByteArrayEntity(compressedBufferData,
+                ContentType.create(prometheusSinkConfiguration.getContentType()), prometheusSinkConfiguration.getEncoding());
+
+        classicHttpRequestBuilder.setEntity(entity);
+        classicHttpRequestBuilder.addHeader("Content-Encoding", prometheusSinkConfiguration.getEncoding());
+        classicHttpRequestBuilder.addHeader("Content-Type", prometheusSinkConfiguration.getContentType());
+        classicHttpRequestBuilder.addHeader("X-Prometheus-Remote-Write-Version", prometheusSinkConfiguration.getRemoteWriteVersion());
+
+        try {
+            if(AuthTypeOptions.BEARER_TOKEN.equals(prometheusSinkConfiguration.getAuthType()))
```

Comment:
How about other Auth support ?

Reply:
@dinujoh , we have unauthenticated,  http_basic, and bearer_token auth just like Http Sink plugin